contracts/src/v0.8/keystone/interfaces/ICapabilityRegistry.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {TypeAndVersionInterface} from "../../interfaces/TypeAndVersionInterface.sol";
+import {OwnerIsCreator} from "../../shared/access/OwnerIsCreator.sol";
+
+interface ICapabilityRegistry {
+    struct NodeOperator {
+        /// @notice Unique identifier for the node operator
+        uint256 id;
+        /// @notice The address of the admin that can manage a node
+        /// operator
+        address admin;
+        /// @notice Human readable name of a Node Operator managing the node
+        string name;
+    }
+
+    struct Capability {
+        // Capability type, e.g. "data-streams-reports"
+        // bytes32(string); validation regex: ^[a-z0-9_\-:]{1,32}$
+        // Not "type" because that's a reserved keyword in Solidity.
+        bytes32 capabilityType;
+        // Semver, e.g., "1.2.3"
+        // bytes32(string); must be valid Semver + max 32 characters.
+        bytes32 version;
+    }
+
+    /// @notice This error is thrown when trying to set a node operator's
+    /// admin address to the zero address
+    error InvalidNodeOperatorAdmin();
+
+    /// @notice This event is emitted when a new node operator is added
+    /// @param nodeOperatorId The ID of the newly added node operator
+    /// @param admin The address of the admin that can manage the node
+    /// operator
+    /// @param name The human readable name of the node operator
+    event NodeOperatorAdded(uint256 nodeOperatorId, address indexed admin, string name);
+
+    /// @notice This event is emitted when a new capability is added
+    /// @param capabilityId The ID of the newly added capability
+    event CapabilityAdded(bytes32 indexed capabilityId);
+
+    /// @notice Adds a new node operator
+    /// @param admin The address of the admin that can manage the node
+    /// operator
+    /// @param name The human readable name of the node operator
+    function addNodeOperator(address admin, string calldata name) external;
```

Comment:
I'm thinking of updating this function signature to accept an array of `NodeOperator` structs so that multiple node operators can be added in a single transaction.  This may be easier if we remove the `id` field from the `NodeOperator` struct.

Reply:
> I'm thinking of updating this function signature to accept an array of NodeOperator structs so that multiple node operators can be added in a single transaction.

👍 makes sense.

> This may be easier if we remove the id field from the NodeOperator struct.

I'm not sure I understand how this makes it easier. Could you elaborate?